ateam_ui/src/src/overlay.ts

```diff
@@ -123,12 +168,57 @@ export class Overlay {
                 break;
             // MESH
             case 6:
-                // I think I can use a PIXI filter to do this more efficiently
+                // This is for 2d arrays of meshes
+                // ASSUMPTION MESH IS IN Y then X or row then column (row-major)
+                // Could just use flat if this was directly an array of arrays
+                let heat_texture = this.array_to_pixi_texture(this.mesh, this.mesh_alpha)
+                // console.log(heat_texture.valid);
+                if (heat_texture.valid) {
+                  this.heat_filter.uniforms.uTexture = heat_texture;
+                //   this.heat_filter.uniforms.uSample = heat_texture;
+
+                  // Filled black rectangle this is mapped to
+                  graphic.beginFill('Black');
+                  graphic.lineStyle(this.stroke_width, 'Black');
+                  graphic.drawRect(-scale*this.scale.x/2, -scale*this.scale.y/2, scale*this.scale.x, scale*this.scale.y);
+                  graphic.endFill();
+                  graphic.filters = [this.heat_filter];
+                }
                 break;
+
             // CUSTOM
             case 7:
                 // TODO: This is probably a very low priority to implement
                 break;
         }
-    }       
+    }
+
+    array_to_pixi_texture(mesh: Mesh1d[], mesh_alpha: Mesh1d[]) {
+      const bpp = 4;
+
+      let height = mesh.length;
+      if (height > 0){
+        let width = mesh[0].mesh1d.length;
+        if (width > 0) {
+          let buff = new Uint8Array(width * height * bpp);
+          for (let i = 0; i < height; i++) {
+            // should do check to make sure each ros is length width
+            if (mesh[i].mesh1d.length != width){break;}
+            for (let j = 0; j < width; j++) {
+              let linear_index = ((width * i) + j) * bpp;
+              buff[linear_index] = mesh[i].mesh1d[j] * 20;
+              buff[linear_index + 1] = 0;
+              buff[linear_index + 2] = 0;
+              buff[linear_index + 3] = 1.0;
+            }
+          }
+
+          return PIXI.Texture.fromBuffer(buff, width, height, {scaleMode: PIXI.SCALE_MODES.LINEAR	});
+        }
+      }
+
+      // Regardless of what we return the user can just check if this.valid is true
+      return PIXI.Texture.fromBuffer(null, 0, 0);
+    }
```

Comment:
It seems like we suddenly switch from 4 space indents to 2 space indents, we should probably keep it consistent